pyvista/plotting/widgets.py

```diff
@@ -140,10 +138,11 @@ def _the_callback(box_widget, event_id):
 
     def clear_box_widgets(self):
         """Disable all of the box widgets."""
-        if hasattr(self, 'box_widgets'):
-            for widget in self.box_widgets:
-                widget.Off()
-            del self.box_widgets
+        # this is necessary for garbage collection
+        for widget in list(self.box_widgets):
```

Comment:
Given that self.box_widgets has type `List[object]`, why is wrapping this in `list` required?

Reply:
Because it creates a copy that can be iterated over meanwhile deleting values from the original list.

What I don't understand is why we can't just do `sefl.box_widgets.clear()`.

Reply:
> What I don't understand is why we can't just do self.box_widgets.clear(). Or self.box_widgets[:] = [] (to replace the whole function with).

I have no idea. It should have garbage collected once the last reference disappeared by overwriting the list. Through trial and error this was the only solution that worked.

Reply:
For completeness' sake: we could, if we remove `self.box_widgets = []` from a few lines lower and always use the class attribute :)